backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -168,10 +197,18 @@
     @RolesAllowed(RbacIdentityProvider.RBAC_WRITE_INTEGRATIONS_ENDPOINTS)
     @APIResponse(responseCode = "200", content = @Content(schema = @Schema(type = SchemaType.STRING)))
     public Uni<Response> updateEndpoint(@Context SecurityContext sec, @PathParam("id") UUID id, @NotNull @Valid Endpoint endpoint) {
+        // This prevents from updating an endpoint from whatever EndpointType to a system EndpointType
+        checkSystemEndpoint(endpoint.getType());
         RhIdPrincipal principal = (RhIdPrincipal) sec.getUserPrincipal();
         endpoint.setAccountId(principal.getAccount());
         endpoint.setId(id);
-        return resources.updateEndpoint(endpoint)
+
+        return resources.getEndpoint(principal.getAccount(), id)
```

Comment:
This has a minor impact on performances, but we could run a query that would only return the `Endpoint` type from its `id` instead of fetching the entire `Endpoint`.

Reply:
That would work.

Reply:
done